services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java

```diff
@@ -36,12 +43,26 @@ public void validateForEdus(ApiToken apiToken) {
    * @param apiToken the ApiToken that needs to be validated.
    */
   public void validateForPpa(ApiToken apiToken) {
-    apiToken.getLastUsedPpac().ifPresent(it -> {
+    apiToken.getLastUsedPpac().ifPresent(getLastUsedEpochSecond -> {
       LocalDate currentDateUtc = TimeUtils.getLocalDateForNow();
-      LocalDate lastUsedForPpaUtc = getLocalDateFor(it);
+      LocalDate lastUsedForPpaUtc = getLocalDateFor(getLastUsedEpochSecond);
+
+      logLastUpdate(getLastUsedEpochSecond, currentDateUtc, lastUsedForPpaUtc);
+
       if (currentDateUtc.isEqual(lastUsedForPpaUtc)) {
         throw new ApiTokenQuotaExceeded();
       }
     });
   }
+
+  private void logLastUpdate(Long getLastUsedEpochSecond, final LocalDate currentDateUtc,
+      final LocalDate lastUsedForPpaUtc) {
+    Instant lastUsedTimeStamp = Instant.ofEpochSecond(getLastUsedEpochSecond);
+    Instant currentTimeStamp = Instant.now();
+    final long diff = ChronoUnit.HOURS.between(lastUsedTimeStamp, currentTimeStamp);
+    if (diff == VALIDITY_IN_HOURS && currentDateUtc.equals(lastUsedForPpaUtc)) {
```

Comment:
must it be exactly 23 hours? or could it even be more?

```suggestion
    if (diff >= VALIDITY_IN_HOURS && currentDateUtc.equals(lastUsedForPpaUtc)) {
```

Reply:
mhm good question. But I think it doesnt matter. It can only be either 23 hours or 24 hours. If it is 24 hours then everything is fine( because then it wouldnt be the same day anymore) thats why I decided to use ==. If it is more than 23 hours then we would also log this when having success.